RELEASE/scripts/autoscend/combat/auto_combat_default_stage2.ash

```diff
@@ -214,18 +214,70 @@ string auto_combatDefaultStage2(int round, monster enemy, string text)
 	}
 	
 	# Instakill handler
-	boolean doInstaKill = true;
-	if($monsters[Lobsterfrogman, Ninja Snowman Assassin] contains enemy)
+	boolean couldInstaKill = true;
+	if($monsters[Smut Orc Pipelayer,Smut Orc Jacker,Smut Orc Screwer,Smut Orc Nailer] contains enemy && get_property("chasmBridgeProgress").to_int() < 30)
+	{
+		//want to do cold damage in stage3
+		if(my_adventures() > 6)
+		{
+			couldInstaKill = false;
+		}
+	}
+	else if($monsters[Lobsterfrogman, Ninja Snowman Assassin] contains enemy)
 	{
 		if(auto_have_skill($skill[Digitize]) && (get_property("_sourceTerminalDigitizeMonster") != enemy))
 		{
-			doInstaKill = false;
+			couldInstaKill = false;
 		}
 	}
 
-	if(instakillable(enemy) && !isFreeMonster(enemy) && doInstaKill)
+	if(instakillable(enemy) && !isFreeMonster(enemy) && couldInstaKill)
 	{
-		if(canUse($skill[lightning strike]))
+		boolean wantInstaKill;
+		
+		boolean waitForDesert;	//free kills can save turns of Ultrahydrated
+		if(get_property("desertExploration").to_int() < 100 && !isActuallyEd())	//need to explore desert
+		{
+			int currentDesertProgressPerTurn = 1 + 
+			(get_property("bondDesert").to_boolean() ? 2 : 0) +
+			(get_property("peteMotorbikeHeadlight") == "Blacklight Bulb" ? 2 : 0) +
+			(my_familiar() == $familiar[Melodramedary] ? 1 : 0) +
+			equipped_amount($item[survival knife]) +
+			equipped_amount($item[UV-resistant compass]) +
+			2 * equipped_amount($item[Ornate Dowsing Rod]);
+			int fightsLeftToExplore = ceil((100 - get_property("desertExploration").to_int()) / currentDesertProgressPerTurn);
+			if(have_effect($effect[Ultrahydrated]) > 0 && have_effect($effect[Ultrahydrated]) < fightsLeftToExplore)
+			{
+				wantInstaKill = true;
+			}
+			else	//near level 11
+			{
+				waitForDesert = my_basestat(my_primestat()) >= 95;
+			}
+		}
+		
+		boolean waitForCyrpt;	//free kills can get more modern zmobies from 1 turn of a double initiative effect in The Defiled Alcove
+		if(get_property("cyrptAlcoveEvilness").to_int() >= 30)	//need to do Alcove. todo: check cyrptEvilBonus() from PR #1114
+		{
+			if(my_location() == $location[The Defiled Alcove] && have_effect($effect[Bow-Legged Swagger]) == 1)
+			{
+				wantInstaKill = true;
+			}
+			else if(auto_have_skill($skill[Bow-Legged Swagger]) && my_basestat(my_primestat()) >= 35 && get_property("_bowleggedSwaggerUsed").to_boolean())
```

Comment:
don't have this skill, but believe need a ! before get_property("_bowleggedSwaggerUsed").to_boolean()